### PR TITLE
Tpetra: Adding Kokkos::deep_copy() counter via KokkosP interface

### PR DESCRIPTION
**Pull request by csiefer2**
Issue: #10968
Tests: Yes
Stakeholder Feedback: n/a

### CODE REVIEW

File: packages/tpetra/core/test/MultiVector/MultiVector_UnitTests.cpp [hunk 1]
```diff
@@ -5202,6 +5203,39 @@ namespace {
...
+    }
+
+    Tpetra::Details::DeepCopyCounter::start();
+    Kokkos::deep_copy(y_h,x_d);
+    size_t count = Tpetra::Details::DeepCopyCounter::stop();   
+
+    TEST_EQUALITY(count,correct_count);
+          
+  }
+
```
Comment (tasmith4) [inline packages/tpetra/core/test/MultiVector/MultiVector_UnitTests.cpp:5237]: would be nice to have tests of reset and get_count.  right now it's not that obvious what the use case for either one is.

Comment (csiefer2) [reply packages/tpetra/core/test/MultiVector/MultiVector_UnitTests.cpp:5237]: Yeah, you're probably right :)

Comment (csiefer2) [reply packages/tpetra/core/test/MultiVector/MultiVector_UnitTests.cpp:5264]: done

Comment (csiefer2) [reply packages/tpetra/core/test/MultiVector/MultiVector_UnitTests.cpp:5264]: Fixed it

**Review by jhux2 — APPROVED** [review]
lgtm

**Review by tasmith4 — DISMISSED** [review]

**Review by jhux2 — APPROVED** [review]
lgtm

**Pull request merged**